acq4/devices/LEDLightSource/LEDLightSource.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+from acq4.devices.Device import *
+from acq4.devices.LightSource import *
+from acq4.devices.DAQGeneric import DAQGeneric, DAQGenericTaskGui
+from PyQt4 import QtCore, QtGui
+import acq4.util.Mutex as Mutex
+
+class LEDLightSource(LightSource):
+    """Simple device which reports the status of the LED Light Sources...reports up to the LightSource object."""
+    
+    def __init__(self, dm, config, name):
+        LightSource.__init__(self, dm, config, name)
+        self.ledconfig = config.get('leds', config)
+
+        self.leds = {}
+
+        for name, conf in self.ledconfig.iteritems():
+            for k, v in conf.iteritems():
+                if (k == "channel"):
+                    chan = v[1]
+                    device = v[0]
+                    dev = dm.getDevice(device)
+
+                    self.leds[name] = (dev, conf['channel'])
+
+        self.state = {}
+
+    def getLEDState(self):
```

Comment:
There is a signal `DAQGeneric.sigHoldingChanged(channel, value)` that you can connect to.